File: Remove Nth Node From End of List/Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int m=1;
        ListNode slow=head;
        ListNode newhead=new ListNode(0);
        ListNode fast=head;
        while(m<n){
            m++;
            fast=fast.next;
        }
        newhead.next=head;
        ListNode prev=newhead;
        while(fast.next!=null){
            prev=prev.next;
            fast=fast.next;
            slow=slow.next;
        }
        if(slow.next!=null){
            slow.val=slow.next.val;
            slow.next=slow.next.next;
        }else{
            prev.next=null;
        }
        return newhead.next;
    }
}

File: Remove Nth Node From End of List/Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @return a ListNode
    def removeNthFromEnd(self, head, n):
        m=1
        fast=head
        while m<n:
            m=m+1
            fast=fast.next
        slow=head
        newhead=ListNode(0)
        newhead.next=head
        prev=newhead
        while fast.next:
            prev=prev.next
            slow=slow.next
            fast=fast.next
        if slow.next:
            slow.val=slow.next.val
            slow.next=slow.next.next
        else:
            prev.next=None
        return newhead.next
